core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java

```diff
@@ -191,8 +189,11 @@ private String toCacheKey(String path, String property, String keyProperty) {
 
     private Stream<Path> resolveAllFromJar(String path) {
         String rootAsString = ResourcePathTypeResolver.class.getProtectionDomain().getCodeSource().getLocation().toString();
+
         ClassLoader classLoader = ObjectHelper.assertNotNull(ResourcePathTypeResolver.class.getClassLoader());
-        if (rootAsString.endsWith(".jar") && !rootAsString.matches(".*" + File.separator + "citrus-api-\\d+\\.\\d+\\.\\d+(-.*)?\\.jar")) {
+        if (rootAsString.matches(".*jar(!/)?") &&
+            !rootAsString.replace("\\", "/")
+                .matches(".*/citrus-api-\\d+\\.\\d+\\.\\d+(-.*)?\\.jar")) {
```

Comment:
hm. I remember @Tom-29 working on this section too. has this resolved his issue as well?